src/webview/webview2/mod.rs

```diff
@@ -311,6 +311,23 @@ impl InnerWebView {
       }
     }
 
+    if let Some(on_load_handler) = attributes.on_load_handler {
+      let window_c = window.clone();
+      unsafe {
+        webview
+          .add_ContentLoading(
```

Comment:
shouldn't we use [`DOMContentLoaded`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.domcontentloaded?view=webview2-dotnet-1.0.1823.32) instead? or [`NavigationCompleted`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.navigationcompleted?view=webview2-dotnet-1.0.1823.32)

Reply:
It depends on how we want to define on_load, [ContentLoading](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.contentloading?view=webview2-dotnet-1.0.1823.32) seemed to most closely match the APIs provided by the other platforms. I would lean towards being as consistent as possible across the platforms.

 This probably does slightly change the definition of on_load as it was being used in tauri triggering an event via loading javascript into the page and that probably does most closely track with DOMContentLoaded. This new definition would be slightly earlier in the load process. on_load in tauri now with it was being triggered by injected javascript is something like server data has been retrieved and loaded, but might not yet be visible. It is currently possible to sometimes navigate to a different page in an on_load callback before the page is displayed. I'm not sure how closely you want to track to the previous behavior.
 
All of the target platform have APIs that allow callbacks on starting to navigate, starting load (which is what this targets) and load complete. If wanted I can additionally add handlers for all three events.

Reply:
Yeah for me personally, I think `on_page_load` would be used to interact with the HTML in the app so I think it should be firedd when one can evaluate a script and access the DOM. cc @lucasfernog, what do you think this API should do?

Reply:
imo handling all 3 events makes the most sense, to me navigation start and load complete are even more important than load start i think. And with that we could name the 3 apis accordingly (or have one api via enums or whatever)

Also fwiw on Windows, NavigationCompleted fires even later than DOMContentLoaded which is what many people expected from tauri's on_page_load

Reply:
I can add all three. Something like on_page_navigation, on_page_loading, on_page_loaded (suggestions on those name would be appreciated). Then should we change on_page_load from tauri to line up with final state?